### ISSUE TITLE
Meta not working as move_key_modifier

### ISSUE THREAD

**Issue by robertdj** [matze/vim-move#15]
Hi,

I'm having problems with the default `move_key_modifier` on both Mac (iTerm/built-in Terminal) and Linux Mint. With

	let g:move_key_modifier = 'C'

everything works as expected, but I've reserved the combinations `C-jk` for other tasks. Manually replacing `s:MoveKey('j')` in 

	execute 'nmap' s:MoveKey('j') '<Plug>MoveLineDown'

with the symbol from (`CTRL+V`) `M-j` also works, but it is a clumsy solution and requires different symbols on Mac and Linux.

The output of `:nmap` shows that the shortcut for `MoveLineDown` is a weird letter, but it doesn't help if I add settings for UTF-8 in a different place in my `vimrc`.

Do you have an idea on what could be wrong?

Thanks,

Robert

**Comment by matze** [comment]
You could try disabling automatic mapping with
```vim
let g:move_map_keys = 0
```
and map them yourself using `<C-v> M-j` et al. I [do the same](https://github.com/matze/configs/blob/47f37330633634fac5657652f3faee6cb3351f1d/.vimrc#L153-L156) (GitHub does not display the escape sequence well) because I also have problems with Ubuntu's gnome-terminal.

**Comment by robertdj** [comment]
Your suggestion worked! 

For future reference I had to clone your config repository to see the escape sequence -- even a raw view did not display the keybinding as `superscript left-bracket`.

On Linux Mint it worked out of the box, but on Mac I had to change the interpretation of the `alt` key:

- In Terminal: Profiles -> Keyboard -> "Use Option as Meta key"
- In iTerm: Profiles -> Keys -> "Left option key acts as" +Esc

Thanks a lot for your help!

**Comment by matze** [comment]
Thanks for the investigation, I added a short notice in the docs about this.

**Comment by ejoubaud** [comment]
FTR, one problem I was having after adding the lines from @matze 's `.vimrc` was that esc would then wait for new keys to be typed. That was especially painful to escape visual mode, where typing `<ESC>k` would move my selection down instead of escaping visual and going down one line.

The only way I was able to solve this was with the [fast keycodes hack](http://vim.wikia.com/wiki/Mapping_fast_keycodes_in_terminal_Vim). My config looks something like this

```
" vim-move config
" for terms that send Alt as Escape sequence
" see http://vim.wikia.com/wiki/Mapping_fast_keycodes_in_terminal_Vim
" for why the <F20> hack. Keeps Esc from waiting for other keys to exit visual
set <F20>=^[j
set <F21>=^[k
vmap <F20> <Plug>MoveBlockDown
vmap <F21> <Plug>MoveBlockUp
nmap <F20> <Plug>MoveLineDown
nmap <F21> <Plug>MoveLineUp
```

Note: Don't just copy/paste this: `^[` in the above should be the escape sequence for escape. You can get (in vim) by typing `<C-v><ESC>` in insert mode. I used regular `^` and `[` here because the textarea won't let me use escape codes but it won't work in your vimrc if you have regular `^[` instead of the escape sequence.

With this, my escape key is back to being snappy to exit visual mode, and I can use `<A-j>` and `<A-k>` for `vim-move` in iterm2 (after setting left and right `option key acts as:` to `+Esc`). (Note: if you use tmux, you'll also want to add `set -s escape-time 0` to `.tmux.conf` for the escape key to be really snappy)

@matze: This issue (and these workarounds) might be worth a "troubleshooting" mention in the readme, linking to the issues. I guess the problem would be encountered by most terminal vim users and they generated 3 issues. I had given up on `vim-move` once before I eventually found the issues. What do you think?

**Comment by matze** [comment]
Thanks for the elaborate investigation. Do you want to open a PR for the change to the README or should I take care of that?

**Comment by kiprasmel** [comment]
@ejoubaud the fix didn't help me, but I went down the rabbit hole to [1] -> [2] -> [3] and [3] finally fixed it!

tl;dr: install `tpope/vim-rsi` ([3])!

[1] https://stackoverflow.com/a/27206531/9285308
[2] https://github.com/vim-utils/vim-alt-mappings#thanks
[3] https://github.com/tpope/vim-rsi
